pretyping/ltac_pretype.ml

```diff
@@ -46,7 +46,8 @@ type extended_patvar_map = constr_under_binders Id.Map.t
 type closure = {
   idents:Id.t Id.Map.t;
   typed: constr_under_binders Id.Map.t ;
-  untyped:closed_glob_constr Id.Map.t }
+  untyped:closed_glob_constr Id.Map.t;
+  genargs : Geninterp.Val.t Id.Map.t; }
```

Comment:
This type is now the same as `ltac_var_map` below, is it useful to have both?

Reply:
I think that one of two should be made opaque and abstract away the implementation details of the classification of variables. For the time being I guess it does not hurt to keep both. (Also there is a nasty recursive type going on, I think it's better to have a toplevel type exporting a non-recursive definition.)